diff -Nru a/node_modules/gatsby/dist/services/initialize.js b/node_modules/gatsby/dist/services/initialize.js
--- a/node_modules/gatsby/dist/services/initialize.js	2022-12-12 10:45:37.504432901 +0200
+++ b/node_modules/gatsby/dist/services/initialize.js	2022-12-12 10:45:37.504432901 +0200
@@ -13,6 +13,8 @@

 var fs = _interopRequireWildcard(require("fs-extra"));

+var _fs = _interopRequireWildcard(require("fs"));
+
 var _mutex = require("gatsby-core-utils/mutex");

 var _md5File = _interopRequireDefault(require("md5-file"));
@@ -444,12 +446,29 @@
   const srcDir = `${__dirname}/../../cache-dir`;
   const siteDir = cacheDirectory;
   const tryRequire = `${__dirname}/../utils/test-require-error.js`;
+  const cacheDirectoryPaths = [];
+  const collectCacheDirectoryPaths = (src, dest) => {
+    cacheDirectoryPaths.push(dest);
+    return true;
+  }

   try {
-    await fs.copy(srcDir, siteDir, {
-      overwrite: true
+    await fs.default.copy(srcDir, siteDir, {
+      filter: collectCacheDirectoryPaths
+    });
+    await fs.default.copy(tryRequire, `${siteDir}/test-require-error.js`, {
+      filter: collectCacheDirectoryPaths
     });
-    await fs.copy(tryRequire, `${siteDir}/test-require-error.js`);
+    // Ensure cacheDirectory to be writable even when srcDir is not
+    for (const cacheDirectoryPath of cacheDirectoryPaths) {
+      try {
+        fs.accessSync(cacheDirectoryPath, _fs.constants.W_OK);
+      } catch (err) {
+        const stat = _fs.statSync(cacheDirectoryPath);
+        const mode = stat.mode | _fs.constants.S_IWUSR;
+        _fs.chmodSync(cacheDirectoryPath, mode);
+      }
+    }
     await fs.ensureDir(`${cacheDirectory}/${lmdbCacheDirectoryName}`); // Ensure .cache/fragments exists and is empty. We want fragments to be
     // added on every run in response to data as fragments can only be added if
     // the data used to create the schema they're dependent on is available.
@@ -576,4 +595,4 @@
     workerPool
   };
 }
-//# sourceMappingURL=initialize.js.map
\ Ei rivinvaihtoa tiedoston lopussa
+//# sourceMappingURL=initialize.js.map
